fmt: accept repeated arguments

BenWiederhake opened the pull request:
Turns out, just a simple `args_override_self(true)` is precisely the correct behavior.

This is also correct with regard to `-p`, `-w`, `-g`; I checked (and found #6352, #6353, #6354 in the process of doing so).

This is work towards #5998.

Note that even though I added some basic tests, a huge chunk of the functionality is still untested, especially `-c`, `-x`/`-X`, `-t`, `-m`, `-q`, `-T`, and of course `-g`, which is broken (known issue).

BenWiederhake commented:
Changes since last push: Forgot to commit the test data, whoops :D

BenWiederhake commented:
Remaining failures are due to #6275 and #6333.

cakebaker commented on tests/by-util/test_fmt.rs:
```diff
@@ -159,3 +185,87 @@ fn test_fmt_set_goal_not_contain_width() {
+#[test]
+fn prefix_minus() {
+    for invocation in [
+        vec!["-p-", "prefixed-one-word-per-line.txt"],
+        vec!["-p", "-", "prefixed-one-word-per-line.txt"],
+        vec!["--prefix=-", "prefixed-one-word-per-line.txt"],
+        vec!["--prefix", "-", "prefixed-one-word-per-line.txt"],
+        vec!["--pref=-", "prefixed-one-word-per-line.txt"],
+        vec!["--pref", "-", "prefixed-one-word-per-line.txt"],
+        // Test self-overriding:
+        vec!["--prefix==", "--prefix=-", "prefixed-one-word-per-line.txt"],
```
To eliminate the duplication of the filename, I suggest to move it to an `arg` call to make it obvious that it's always the same filename: 
```rust
ucmd.args(&invocation)
    .arg("prefixed-one-word-per-line.txt")
    .succeeds()
    .stdout_is(at.read("prefixed-one-word-per-line_p-.txt"));
```
The same applies to the tests below.

BenWiederhake replied:
Oh! So obvious, yet I didn't notice it. Yes, thanks!

cakebaker commented on tests/by-util/test_fmt.rs:
```diff
@@ -159,3 +185,87 @@ fn test_fmt_set_goal_not_contain_width() {
+        ucmd.arg("one-word-per-line.txt")
+            .arg(arg)
+            .succeeds()
+            .stdout_is(at.read("one-word-per-line.txt"));
```
You can simplify this by using `stdout_is_fixture("one-word-per-line.txt");`. This also applies to the tests below.

BenWiederhake commented:
Changes since last push:
- rebased
- extracted common arg to common `arg()` (thanks @cakebaker!)
- use `stdout_is_fixture` as appropriate (thanks @cakebaker!)

BenWiederhake commented:
Changes since last push:
- Now that `at_and_ucmd` is no longer needed, don't use it. (duh)

cakebaker commented:
Thanks!
